src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -0,0 +1,300 @@
+            base(classType, propertyType, propertyInfo, elementType, options)
+        {
+            if (propertyInfo != null)
+            {
+                Set = Options.ClassMaterializerStrategy.CreateSetter<TValue>(propertyInfo);
+                Get = Options.ClassMaterializerStrategy.CreateGetter<TValue>(propertyInfo);
+            }
+            else
+            {
+                // Used when current.obj contains the return value and this is the property policy.
+                Get = delegate (object obj)
+                {
+                    return (TValue)obj;
+                };
+            }
+
+            GetPolicies();
+        }
+
+        public override Type TempListType
+        {
+            get
+            {
+                return typeof(List<TValue>);
+            }
+        }
+
+        public override object GetValueAsObject(object obj)
+        {
+            Debug.Assert(Get != null);
+            return Get(obj);
+        }
+
+        public override void SetValueAsObject(object obj, object value)
+        {
+            Debug.Assert(Set != null);
+            Set(obj, (TValue)value);
+        }
+
+        public override PropertyValueConverterAttribute GetValueConverter()
+        {
+            return ValueConverter;
+        }
+
+        public override void GetPolicies()
+        {
+            // ValueConverter
+            ValueConverter = GetPropertyValueConverter<TValue>();
+
+            base.GetPolicies();
+        }
+
+        protected PropertyValueConverterAttribute GetPropertyValueConverter<TProperty>()
+        {
+            Type propertyType = typeof(TProperty);
+
+            PropertyValueConverterAttribute attr = null;
+            if (PropertyInfo != null)
+            {
+                // Use Property first
+                attr = Options.GetAttributes<PropertyValueConverterAttribute>(PropertyInfo).Where(a => a.PropertyType == propertyType).FirstOrDefault();
+            }
+
+            if (attr == null)
+            {
+                // Then class type
+                attr = Options.GetAttributes<PropertyValueConverterAttribute>(ClassType, inherit: true).Where(a => a.PropertyType == propertyType).FirstOrDefault();
+                if (attr == null)
+                {
+                    // Then declaring assembly
+                    attr = Options.GetAttributes<PropertyValueConverterAttribute>(ClassType.Assembly).Where(a => a.PropertyType == propertyType).FirstOrDefault();
+
+                    if (attr == null)
+                    {
+                        // Then default
+                        if (propertyType == typeof(DateTime))
+                        {
+                            attr = s_dateConverterAttibute;
+                        }
+                        else if (propertyType == typeof(Enum))
+                        {
+                            attr = s_enumConverterAttibute;
+                        }
```

Comment:
This code could be simplified too. If you would like I can submit a PR on the weekend to rewrite such things.